### ISSUE TITLE
Select component is not accessible. Does not announce to screen readers properly.

### ISSUE THREAD

**Issue by douglasbouttell-camunda** [bpmn-io/form-js#1066]
### Describe the Bug

When using the Select component with a screenreader (for example, voiceover on Mac) it does not correctly announce the content and options to select.

### Steps to Reproduce
Use a default select component.

### Expected Behavior
| Situation | What is currently announced | What should be announced[^1] |
|---|---|---|
| Select top element | "_VALUE_, contents selected, _LABEL_, edit text" | "_VALUE_, _LABEL_, list box pop-up, Menu pop-up combo box" |
| Open dropdown | "_VALUE_, contents selected, _LABEL_, edit text" (note: dropdown opens on focus) | "_VALUE_, selected, (_x_ of _y_), completion selected" |
| Navigate values | The currently chosen value | "_VALUE_, not selected, (_x_ of _y_), completion selected" |

[^1]: based off [WCAG recommendations for comboboxes](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/)

Additionally, since this is modelled as a text field, keyboard navigation will highlight this field allowing for the value to be changed to anything.

### What to do
Follow the example given by W3C https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/ and adjust styles to match.

### Environment

- Host (Browser/Node version), if applicable: Chrome 121.0.6167.184
- OS: MacOS 14.3
- Library version: viewer 1.7.0

**Comment by douglasbouttell-camunda** [comment]
WIP in https://github.com/bpmn-io/form-js/tree/new-select-component

**Comment by barmac** [comment]
@douglasbouttell-camunda Do you plan to finish this for tomorrow's code freeze? I am asking because it seems to be required for https://github.com/camunda/product-hub/issues/2189

**Comment by douglasbouttell-camunda** [comment]
Hi @barmac. As this was a side-of-desk project and I am departing the company I do not have any plans to complete this. The branch mentioned previously has my progress, which is very advanced. I would say it just needs testing and validation.

**Comment by crobbins215** [comment]
@douglasbouttell-camunda / @marcosgvieira / @volodymyr-melnykc  can this be reassigned to someone on the HTO team?

**Comment by volodymyr-melnykc** [comment]
I've reassigned this issue to @Skaiir
